clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/MigrateLoadBalancerStrategy.java

```diff
@@ -0,0 +1,452 @@
+   * @return the result set
+   */
+  public MigrateLoadBalancerResult generateResults(SecurityGroupLookup sourceLookup, SecurityGroupLookup targetLookup,
+                                                    LoadBalancerLocation source, LoadBalancerLocation target,
+                                                    String subnetType, String applicationName, boolean dryRun) {
+
+    this.sourceLookup = sourceLookup;
+    this.targetLookup = targetLookup;
+
+    final MigrateLoadBalancerResult result = new MigrateLoadBalancerResult();
+
+    LoadBalancerDescription sourceLoadBalancer = getLoadBalancer(source.getCredentials(), source.getRegion(), source.getName());
+    Vpc sourceVpc = getVpc(source);
+    Vpc targetVpc = getVpc(target);
+
+    String targetName = generateLoadBalancerName(source.getName(), sourceVpc, targetVpc);
+    LoadBalancerDescription targetLoadBalancer = getLoadBalancer(target.getCredentials(), target.getRegion(), targetName);
+
+    List<MigrateSecurityGroupResult> targetGroups = getTargetSecurityGroups(sourceLoadBalancer, source, target);
+
+    List<String> securityGroups = targetGroups.stream().map(g -> g.getTarget().getTargetName()).distinct().collect(Collectors.toList());
+    securityGroups.addAll(buildExtraSecurityGroups(source, target, sourceLoadBalancer, applicationName, result, dryRun));
+
+    result.getSecurityGroups().addAll(targetGroups);
+    result.setTargetName(targetName);
+    result.setTargetExists(targetLoadBalancer != null);
+    if (!dryRun) {
+      updateTargetLoadBalancer(source, target, sourceLoadBalancer, targetLoadBalancer, targetName, subnetType, securityGroups);
+    }
+
+    return result;
+  }
+
+  /**
+   * Performs the actual upsert operation against the target load balancer
+   *
+   * @param source             the source load balancer
+   * @param target             the target location
+   * @param sourceLoadBalancer the Amazon load balancer description of the source load balancer
+   * @param targetLoadBalancer the Amazon load balancer description of the target load balancer (may be null)
+   * @param targetName         the name of the target load balancer
+   * @param subnetType         the subnetType in which to migrate the load balancer (should be null for EC Classic migrations)
+   * @param securityGroups     a list of security group names to attach to the load balancer
+   */
+  protected void updateTargetLoadBalancer(LoadBalancerLocation source, LoadBalancerLocation target,
+                                        LoadBalancerDescription sourceLoadBalancer,
+                                        LoadBalancerDescription targetLoadBalancer,
+                                        String targetName, String subnetType, Collection<String> securityGroups) {
+
+    List<Listener> listeners = sourceLoadBalancer.getListenerDescriptions().stream()
+      .map(ListenerDescription::getListener)
+      .filter(l -> l.getSSLCertificateId() == null ||
```

Comment:
I'm a bit confused on the filtering bit here

Do we just ignore a listener that has an associated cert if we are migrating it?
If so it seems like that should come back as a migration warning